modules/math/shared/src/main/scala/gsp/math/skycalc/solver/TargetCalculator.scala

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gsp.math.skycalc.solver
+
+import java.time.Duration
+import java.time.Instant
+import gsp.math.Coordinates
+import gsp.math.Place
+import gsp.math.skycalc.ImprovedSkyCalc
+import gsp.math.skycalc.SkyCalcResults
+
+/**
+  * Target calculator that allows to calculate different attributes of a target for a given set of instants.
+  *
+  * If in doubt use {@link isDefinedAt} to make sure that values for a given time are actually calculated before
+  * accessing them, otherwise an out of bounds exception will be thrown.
+  */
+trait TargetCalculator[G] extends Calculator[G, SkyCalcResults] {
+
+  val place: Place
```

Comment:
Yes, I introduced this when I ported `SkyCalc` since `Site` is defined in `gsp-core`. If we end up moving `Enumerated` and `Site` to this library then `Place` may not be necessary. Or a `Site` will be a `Place`.